File: 0357-count-numbers-with-unique-digits/0357-count-numbers-with-unique-digits.js
/**
 * @param {number} n
 * @return {number}
 */
var countNumbersWithUniqueDigits = function(n) {
    let total = 10
    let curr = 9
    let start = 9
    if (n === 0) return 1
    while(n > 1 && curr > 0){
        curr *= start
        total += curr
        start -=1
        n --
    }
    return total
    
};

File: 0357-count-numbers-with-unique-digits/0357-count-numbers-with-unique-digits.py
class Solution:
    def countNumbersWithUniqueDigits(self, n: int) -> int:
        if n==0: return 1
        total = 10
        start = 9
        current = 9
        while n>1 and start >0:
            current *= start
            total += current
            start -=1
            n -=1
        return total
        